packages/link-modules/src/services/link-module-service.ts

```diff
@@ -0,0 +1,91 @@
+import { PivotService } from "@services"
+import {
+  Context,
+  DAL,
+  FindConfig,
+  InternalModuleDeclaration,
+  ModuleJoinerConfig,
+  ProductTypes,
+} from "@medusajs/types"
+import { InjectTransactionManager, MedusaContext } from "@medusajs/utils"
+import { shouldForceTransaction } from "../utils"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+  pivotService: PivotService<any>
+}
+
+export default class LinkModuleService<TPivot> {
+  baseRepository_: DAL.RepositoryService
+  readonly pivotService_: PivotService<TPivot>
+
+  constructor(
+    { baseRepository, pivotService }: InjectedDependencies,
+    readonly moduleDeclaration: InternalModuleDeclaration
+  ) {
+    this.baseRepository_ = baseRepository
+    this.pivotService_ = pivotService
+  }
+
+  __joinerConfig(): ModuleJoinerConfig {
+    return {} as ModuleJoinerConfig
+  }
+
+  async list(
+    filters: ProductTypes.FilterableProductProps = {},
+    config: FindConfig<unknown> = {},
+    sharedContext?: Context
+  ): Promise<ProductTypes.ProductDTO[]> {
+    const products = await this.pivotService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return JSON.parse(JSON.stringify(products))
+  }
+
+  async listAndCount(
+    filters: ProductTypes.FilterableProductProps = {},
+    config: FindConfig<unknown> = {},
+    sharedContext?: Context
+  ): Promise<[ProductTypes.ProductDTO[], number]> {
+    const [products, count] = await this.pivotService_.listAndCount(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return [JSON.parse(JSON.stringify(products)), count]
+  }
+
+  @InjectTransactionManager(shouldForceTransaction, "baseRepository_")
+  async create(data: unknown[], @MedusaContext() sharedContext: Context = {}) {
```

Comment:
todo: create a protected method such as `create_` that is wrapped in a transaction and call it here, then call the serialize from the base repository on the result. Otherwise here we are returning a mikro orm object